inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/IcebergMultipleStreamWriter.java

```diff
@@ -201,7 +238,41 @@ public void processElement(RecordWithSchema recordWithSchema) throws Exception {
 
         if (multipleWriters.get(tableId) != null) {
             for (RowData data : recordWithSchema.getData()) {
-                multipleWriters.get(tableId).processElement(data);
+                try {
+                    multipleWriters.get(tableId).processElement(data);
+                } catch (Exception e) {
+                    LOG.error(String.format("write error, raw data: %s", data), e);
+                    if (!dirtyOptions.ignoreDirty()) {
+                        throw e;
+                    }
+                    if (dirtySink != null) {
+                        DirtyData.Builder<Object> builder = DirtyData.builder();
+                        try {
+                            builder.setData(data)
+                                    .setLabels(dirtyOptions.getLabels())
+                                    .setLogTag(dirtyOptions.getLogTag())
+                                    .setIdentifier(dirtyOptions.getIdentifier())
+                                    .setRowType( multipleWriters.get(tableId).getFlinkRowType())
+                                    .setDirtyMessage(e.getMessage());
+                            dirtySink.invoke(builder.build());
+                            if (sinkMetricData != null) {
+                                sinkMetricData.invokeDirtyWithEstimate(data);
+                            }
```

Comment:
here should use table level dirty data method

Reply:
use this method to get more accurate metrics : metricdata.outputdirtymetricswithestimate()